src/crimson/osd/pg_backend.cc

```diff
@@ -528,12 +528,13 @@ seastar::future<> PGBackend::omap_get_keys(
       bool truncated = false;
       uint32_t num = 0;
       for (auto& [key, val] : std::get<1>(ret)) {
-        if (num++ >= max_return ||
+        if (num >= max_return ||
```

Comment:
a minor eye-sore: we are collecting the size in a uint32_t, then verify it against a uint64_t max...
(I guess the reason is the decoding - but why was it decoded in uint64_t?)

Reply:
It think it's legacy. The direct reason is how `encoding.h` handles containers. The number of items if always conveyed using 32 bits:

```cpp
// std::set<T>
template<class T, class Comp, class Alloc, typename traits>
inline std::enable_if_t<!traits::supported>
  encode(const std::set<T,Comp,Alloc>& s, bufferlist& bl)
{
  __u32 n = (__u32)(s.size());
  encode(n, bl);
  for (auto p = s.begin(); p != s.end(); ++p)
    encode(*p, bl);
}
```